src/dune_pkg/rev_store.ml

```diff
@@ -493,13 +511,14 @@ module Entry = struct
       |> Option.bind ~f:(fun m ->
         match Re.Group.get m 1 with
         | "blob" ->
-          Some
-            (File
-               (Direct
-                  { hash = Re.Group.get m 2
-                  ; size = Int.of_string_exn @@ Re.Group.get m 3
-                  ; path = Path.Local.of_string @@ Re.Group.get m 4
-                  }))
+          let open Option.O in
+          let+ hash = Re.Group.get m 2 |> Object.of_sha1 in
```

Comment:
Can this return None or does the regex prevent it?

Reply:
Good point, it can't happen.